VultisigApp/VultisigApp/Services/SwapService/SwapService.swift

```diff
@@ -11,15 +11,42 @@ struct SwapService {
 
     static let shared = SwapService()
 
-    private let thorchainService: ThorchainService = ThorchainService.shared
+    private let thorchainService: ThorchainSwapProvider = ThorchainService.shared
+    private let mayachainService: ThorchainSwapProvider = MayachainService.shared
     private let oneInchService: OneInchService = OneInchService.shared
 
     func fetchQuote(amount: Decimal, fromCoin: Coin, toCoin: Coin, isAffiliate: Bool) async throws -> SwapQuote {
-        guard let fromChainID = fromCoin.chain.chainID, let toChainID = toCoin.chain.chainID, fromChainID == toChainID else {
-            return try await fetchThorchainQuote(amount: amount, fromCoin: fromCoin, toCoin: toCoin, isAffiliate: isAffiliate)
+        
+        // 1Inch resolver
+        if let fromChainID = fromCoin.chain.chainID,
+           let toChainID = toCoin.chain.chainID, fromChainID == toChainID
+        {
+            return try await fetchOneInchQuote(
+                chain: fromChainID,
+                amount: amount, fromCoin: fromCoin,
+                toCoin: toCoin, isAffiliate: isAffiliate
+            )
+        }
+
+        // Mayachain resolver
+        if fromCoin.chain == .mayaChain || toCoin.chain == .mayaChain {
```

Comment:
I don't think this check is correct , on MayaChain , you can swap from RUNE<->CACAO, but you can also swap Between KUJI<-> Dash , if you trying to swap between KUJI <-> DASH , this will not able to get a quote

Reply:
We'll toggle KUJI<-> Dash swaps in next PR. I think PR is already big. I'm still not sure that it's working need to do separate testing and may be fix something. In current PR is only RUNE->CACAO  CACAO->RUNE